realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTests.java

```diff
@@ -737,6 +694,17 @@ public void run() {
                 @Override
                 public void onChange(Dog object) {
                     typebasedCommitInvocations.incrementAndGet();
+                    if (typebasedCommitInvocations.get() > NUMBER_OF_LISTENERS) {
+                        fail();
+                    } else if (typebasedCommitInvocations.get() == NUMBER_OF_LISTENERS) {
+                        // Delayed post in case the listener gets triggered more time than expected.
```

Comment:
we post call of `parent.setFieldObject(child);` just in case the listener gets called more than `NUMBER_OF_LISTENERS` times. In that case, it will fail on line 698.